common/src/main/java/me/shedaniel/architectury/event/events/PlayerEvent.java

```diff
@@ -70,62 +105,179 @@
+         * @param smelted The smelt result.
+         */
         void smelt(Player player, ItemStack smelted);
     }
     
     interface PickupItemPredicate {
+        /**
+         * Invoked when a player tries to pickup an {@link ItemEntity}.
+         * Equivalent to Forge's {@code EntityItemPickupEvent} event.
+         *
+         * @param player The player picking up.
+         * @param entity The {@link ItemEntity} that the player tries to pick up.
+         * @param stack  The content of the {@link ItemEntity}.
+         * @return Forge ignores the return value. On Fabric a {@link InteractionResult#FAIL} results in the event getting canceled and the item not being picked up.
+         */
         InteractionResult canPickup(Player player, ItemEntity entity, ItemStack stack);
     }
     
     interface PickupItem {
+        /**
+         * Invoked when a player has picked up an {@link ItemEntity}.
+         * Equivalent to Forge's {@code ItemPickupEvent} event.
+         *
+         * @param player The player.
+         * @param entity The {@link ItemEntity} that the player picked up.
+         * @param stack  The content of the {@link ItemEntity}.
+         */
         void pickup(Player player, ItemEntity entity, ItemStack stack);
     }
     
     interface ChangeDimension {
+        /**
+         * Invoked when a player changes their dimension.
+         * Equivalent to Forge's {@code PlayerChangedDimensionEvent} event.
+         *
+         * @param player   The teleporting player.
+         * @param oldLevel The level the player comes from.
+         * @param newLevel The level the player teleports into.
+         */
         void change(ServerPlayer player, ResourceKey<Level> oldLevel, ResourceKey<Level> newLevel);
     }
     
     interface DropItem {
+        /**
+         * Invoked when a player drops an item.
+         * Equivalent to Forge's {@code ItemTossEvent} event.
+         *
+         * @param player The player dropping something.
+         * @param entity The entity that has spawned when the player dropped a ItemStack.
+         * @return Forge ignores the return value. On Fabric a {@link InteractionResult#FAIL} results in the event getting canceled and the item not being dropped.
+         */
         InteractionResult drop(Player player, ItemEntity entity);
     }
     
     interface BreakBlock {
+        /**
+         * Deprecated variant of {@link BlockEvent#BREAK}. Defers to the proper event.
+         */
         InteractionResult breakBlock(Level world, BlockPos pos, BlockState state, ServerPlayer player, @Nullable IntValue xp);
     }
     
     interface OpenMenu {
+        /**
+         * Invoked when a player opens a menu.
+         * Equivalent to Forge's {@code PlayerContainerEvent.Open} event.
+         *
+         * <p>If you want to prevent a certain menu to be opened, you have to catch the {@link me.shedaniel.architectury.event.events.GuiEvent#SET_SCREEN} event and open your own.
```

Comment:
This may cause a server client desync, we should look into making this event cancellable instead.